### DISCUSSION TITLE
DSG trajectory of drifters measuring parameters at different depth

### DISCUSSION THREAD

**Discussion by ludo-ifr** [cf-convention/discuss#368] [Q&A about using CF]
Dear cf-convention

I would like to know how to manage drifters that are measuring different parameters at different depth.

Example : We provide the drifter trajectory with:

-  wind (from model) at depth = 0m, 
- sea temperature measured at -0.5 m, 
- and current measured at -15m

First: Do you think it is a trajectoryProfile or a trajectory ?
As you explain that a profile is a series of connected observations along a vertical line, I would say that this is a trajectory

If it is a trajectory, could I used this representation:

````
dimensions:
      obs = 1000 ;
      level = 3 ;

   variables:
      string trajectory ;
        trajectory:cf_role = "trajectory_id";
        trajectory:long_name = "trajectory name" ;
      float depth(obs, level) ;
           depth:axis = "Z" ;
      float temp(obs,level) ;
           temp:coordinates = "time lon lat depth" ;
      float current(obs,level) ;
           current:coordinates = "time lon lat depth" ;
      float wind(obs,level) ;
           wind:coordinates = "time lon lat depth" ;
````

The problem of such representation (incomplete multidimensional array representation) is the sacrifice of storage (but it is acceptable)

I was also thinking that it could be 3 trajectories (in the same file) :

-  One for the wind
-  one for the temperature,
-  one for the current

I have some others questions about the depth depending on level:
Is it possible to have a missing value for depth (I think it is acceptable as it is an auxialiary coordinate) ?
Should the depth be ordered monotonically ?

Currently we could have in our files:
temp= 
_, 10.0 , _,
11.0, _, _

and the associated depth:
depth=
0, -0.5, 15,
-0.5, 15, _

This is correct as the temperature is always at -0.5m but it is difficult to use for user.

Thanks for your help

**Comment by JonathanGregory** [comment]
Dear @ludo-ifr 

I have converted your question from an issue to a Q&A discussion (because we have changed the way questions are handled). I am sorry that no-one answered you when you asked the question 18 months ago. I expect that you have already answered it for yourself long ago.

In case it's still relevant, I would agree with you that this is a trajectory.  Table 9.1 says that a trajectory has `data(i,o)` and must have coordinates  `x(i,o)`, `y(i,o)` and `t(i,o)`, where `i` is the instance and `o` the element (observation). The vertical coordinate is not required, and therefore could be supplied as a scalar for each quantity, which I think would suit your need:

```
dimensions:
  obs = 1000 ;
variables:
  string trajectory ;
    trajectory:cf_role = "trajectory_id";
    trajectory:long_name = "trajectory name" ;
  float time(obs);
    time:units="seconds since 2024-9-20";
  float lon(obs);
    lat:units="degrees_east";
  float lat(obs);
    lat:units="degrees_north";
  float temp(obs) ;
    temp:coordinates = "time lon lat depthtemp" ;
    temp:standard_name="sea_water_temperature";
    temp:units="degC";
  float depthtemp;
    depthtemp:standard_name="depth";
    depthtemp:units="m";
  float current(obs,level) ;
    current:coordinates = "time lon lat depthcurrent" ;
    current:standard_name="sea_water_speed";
    current:units="m s-1";
  float depthcurrent;
    depthcurrent:standard_name="depth";
    depthcurrent:units="m";
  float wind(obs,level) ;
    wind:coordinates = "time lon lat depthwind" ;
    wind:standard_name="wind_speed";
    wind:units="m s-1";
  float depthwind;
    depthwind:standard_name="depth";
    depthwind:units="m";
data:
  depthtemp=-0.5;
  depthcurrent=-15;
  depthwind=0;

Best wishes

Jonathan
